hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ReconTaskControllerImpl.java

```diff
@@ -173,21 +178,27 @@ public synchronized void reInitializeTasks(
       for (Map.Entry<String, ReconOmTask> taskEntry :
           reconOmTasks.entrySet()) {
         ReconOmTask task = taskEntry.getValue();
+        taskStatusMap.get(task.getTaskName()).setCurrentTaskRunStatus(1);
         tasks.add(() -> task.reprocess(omMetadataManager));
       }
       List<Future<Pair<String, Boolean>>> results =
           executorService.invokeAll(tasks);
       for (Future<Pair<String, Boolean>> f : results) {
         String taskName = f.get().getLeft();
+        ReconTaskStatus reconTaskStatusRecord = taskStatusMap.get(taskName);
+        reconTaskStatusRecord.setLastUpdatedSeqNumber(omMetadataManager.getLastSequenceNumberFromDB());
+        reconTaskStatusRecord.setLastUpdatedTimestamp(System.currentTimeMillis());
         if (!f.get().getRight()) {
           LOG.info("Init failed for task {}.", taskName);
+          reconTaskStatusRecord.setLastTaskRunStatus(-1);
+          taskStatusCounter.updateCounter(taskName, false);
         } else {
           //store the timestamp for the task
-          ReconTaskStatus reconTaskStatusRecord = new ReconTaskStatus(taskName,
-              System.currentTimeMillis(),
-              omMetadataManager.getLastSequenceNumberFromDB());
-          reconTaskStatusDao.update(reconTaskStatusRecord);
+          reconTaskStatusRecord.setLastTaskRunStatus(1);
```

Comment:
Correct the status code.

Reply:
Addressed in [90c1997](https://github.com/apache/ozone/pull/7517/commits/90c1997d1a1c2906198d2117542ba2fa10869a56)